executor/table_reader.go

```diff
@@ -114,11 +120,19 @@ func (e *TableReaderExecutor) Open(ctx context.Context) error {
 		defer span1.Finish()
 		ctx = opentracing.ContextWithSpan(ctx, span1)
 	}
-
 	e.memTracker = memory.NewTracker(e.id, -1)
 	e.memTracker.AttachTo(e.ctx.GetSessionVars().StmtCtx.MemTracker)
 
 	var err error
+	defer func() {
+		if e.storeType == kv.TiFlash {
+			if err != nil {
+				TiFlashExecuteErrorCounter.Inc()
+				return
+			}
+			TiFlashExecuteSuccCounter.Inc()
+		}
+	}()
```

Comment:
Collect metrics in here is not right:
1. `TableReaderExecutor.Open` return no error doesn't indicate the query is executed successfully.
2. If the query has many `TableReaderExecutor`, such as `select * from tiflash_table_1 join tiflash_table_2`, it will record 2 metrics for this 1 query.